bitcoin/src/blockdata/weight.rs

```diff
@@ -31,30 +31,38 @@ impl Weight {
     /// Maximum possible value.
     pub const MAX: Weight = Weight(u64::MAX);
 
+    /// The factor that non-witness serialization data is multiplied by during weight calculation.
+    pub const WITNESS_SCALE_FACTOR: u64 = crate::blockdata::constants::WITNESS_SCALE_FACTOR as u64;
+
     /// The maximum allowed weight for a block, see BIP 141 (network rule).
     pub const MAX_BLOCK: Weight = Weight(4_000_000);
 
     /// The minimum transaction weight for a valid serialized transaction.
-    pub const MIN_TRANSACTION: Weight = Weight(4 * 60);
+    pub const MIN_TRANSACTION: Weight = Weight(Self::WITNESS_SCALE_FACTOR * 60);
 
     /// Directly constructs `Weight` from weight units.
     pub const fn from_wu(wu: u64) -> Self { Weight(wu) }
 
-    /// Constructs `Weight` from kilo weight units returning `None` if overflow occurred.
+    /// Directly constructs `Weight` from usize weight units.
+    pub const fn from_wu_usize(wu: usize) -> Self { Weight(wu as u64) }
```

Comment:
Don't stop the PR for me but FTR I don't like this, its hackish, why not `from_wu_u32` as well? I know @apoelstra suggested it and I should have said something at the time. I came up with a possible nicer solution in #2024.

Reply:
Do you mean place `from_wu_u32` instead of `from_wu_usize`?  depending on the arch that this is compiled on, usize could be u32 or u64..

Anyway, if I don't get to it this PR, I'm happy to make this change in a future on.